versionutils/version.go

```diff
@@ -18,6 +18,11 @@ const (
 	SemverMinimumVersion  = "v0.0.1"
 )
 
+// SpeciallyOrderedPrefixes are prefixes that are ordered in a special way
+// This is exported so that consuming packages can specify their own special labels
+// These formats take precedence over alphanumeric ordering for labels
+var SpeciallyOrderedPrefixes = []string{"rc", "beta", "dev"}
```

Comment:
I think this may be in reverse order? It's confusing to think about because "less" actually means lower priority, right? As a dev release will occur before a beta?
I believe we'd want: "-bogus0" < "-dev0" < "-beta0" < "-rc0".
So we need update the logic so that unrecognized prefixes are the least and then this list is reversed or considered as most to least.

Reply:
This list as is will put priority on the first argument. Ie rc is latest

Reply:
Added a unit test to confirm. Looks good.